### DISCUSSION TITLE
What is the `clockwork` directory, why is it read-protected, and does it matter for backups?

### DISCUSSION THREAD

**Discussion by maples3** [LycheeOrg/Lychee#2528] [Q&A] [answered]
As the title states: what is this directory used for?

I have a few Lychee instances that I run backups for, and everything works fine except for permission errors on the `clockwork` directory:
```
.../Lychee/storage/clockwork: dir_open: [Errno 13] Permission denied: 'clockwork'
```

I've looked at this directory, and it's empty in all of my instances, which leads me to believe I can ignore this error.  I've been ignoring it for quite some time, but figured I should get a solid idea of what's going on to see if these backups are actually valid :)

I've tried setting the permissions so my backup-reader user can access it, but the permissions always get reset again (at least on upgrades), which leads me to believe that some process intentionally wants to keep this directory private.  But since it's empty, I don't see the point...?

I looked up the Composer package (https://github.com/itsgoingd/clockwork), and it looks like "clockwork" refers to a development tool, so I'm not sure why it's included in the Lychee builds (if this is indeed the correct "clockwork").

Any tips would be appreciated!

**Comment by ildyria** [comment, marked as answer]
Hi, currently on vacations outside of internet range, not touching lychee at the moment.

Clockwork is indeed development stuff. No need to backup. :)

⁣Get BlueMail for Android ​

**Reply by maples3** [reply]
Awesome, thank you!
